src/oracle/OracleWriter.sol

```diff
@@ -1,16 +1,56 @@
 pragma solidity 0.8.13;
 
-import "src/interfaces/IOracleWriter.sol";
+import { IOracleWriter } from "src/interfaces/IOracleWriter.sol";
+import { LogCompression } from "src/libraries/LogCompression.sol";
+import { stdMath } from "forge-std/Test.sol";
 
 abstract contract OracleWriter is IOracleWriter {
+    // 10 basis points per second which is 6% per minute and doubling of the price in 16 minutes
+    uint256 internal constant MAX_CHANGE_PER_SEC = 0.001e18;
```

Comment:
I think this is quite a high rate of change but still, to set it as a  compile time max seems like we would want to set it even higher in case there is ever such a use case?

Reply:
so do you propose we do not have an max absolute (constant) cap, like we do for the swap / platform fees?

Reply:
I can see the value in catching fat fingers but this can never be circumvented either. When we set the clamping to a really high number it doesnt insta rug it just behaves like UniV3 which is theoretically vulnerable under certain conditions.

Is there a number we can choose that protects against 2 or 3 block attacks but still gives us huge wiggle room. E.g. if 24-36 seconds is a 3 block attack then can we do like a 5% move every 36 seconds? Protects fat fingers, nullifies 3 block attacks but still gives us huge range?

Reply:
I guess 5% per 36 seconds is close to 6% per minute. Maybe your value is already quite good...

Reply:
We can set this to a pretty high number and still achieve have the desired mitigating effect. Because for the attackers to successfully manipulate the TWAP meaningfully, they have to manipulate the spot prices by A LOT during those 5 blocks. 

So imo, there are two ways to derive this threshold:
1. empirically, we can look at real DEX data, and then pick a threshold that covers 99.9% (or whatever %) of cases. Kind of like hypothesis testing in statistics
2. by feeling. If we set it to `0.01e18` (10X the current rate), then in the case of a 5-block attacker, he can move the clamped price `[12%, 24%, ... , 60%]` in 1 minute. In the grand scheme of a 30min TWAP, maybe he moved it 5%

Reply:
I propose 10x to keep it simple & done

Reply:
okay I agree. done.